index.ts

```diff
@@ -45,5 +47,49 @@ export function patternToRegex(...matchPatterns: readonly string[]): RegExp {
 		return allStarsRegex;
 	}
 
-	return new RegExp(matchPatterns.map(x => getRawRegex(x)).join('|'));
+	return new RegExp(matchPatterns.map(x => getRawPatternRegex(x)).join('|'));
+}
+
+// The parens are required by .split() to preserve the symbols
+// Captures consecutive wildcards as one
+const globSymbols = /([?]|[*]+)/;
+function splitReplace(part: string, index: number) {
+	if (part === '') {
+		return ''; // Shortcut for speed
+	}
+
+	if (index % 2 === 0) {
+		// Raw text, escape it
+		return escapeStringRegexp(part);
+	}
+
+	// Else: Symbol
+	if (part.includes('*')) { // Can be more than one
+		return '.*';
+	}
+
+	if (isFirefox) {
+		return '.';
+	}
+
+	return '.?';
+}
+
+function getRawGlobRegex(glob: string): string {
+	return glob
+		.split(globSymbols)
+		// eslint-disable-next-line unicorn/no-array-callback-reference -- tis ok 🤫
+		.map(splitReplace)
+		.join('')
+		.replace(/^([.][*])?/, match => match === '' ? '^' : '')
+		.replace(/([.][*])?$/, match => match === '' ? '$' : '');
```

Comment:
i dont really understand why these two replaces are necessary is it just for perfomance? but everything else seems fine

Reply:
They're just meaningless sequences, it's a regex optimization. 

Since include_glob and exclude_glob are not meant to be used with `.match`, `/^.+a.+$/` is identical to `/a/` when testing URLs. 

I rewrote this piece of code 10 times and hopefully my next commit made it better. 😅

Reply:
yeah seems better